stable/kong/templates/deployment.yaml

```diff
@@ -28,6 +28,10 @@ spec:
         image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
         imagePullPolicy: {{ .Values.image.pullPolicy }}
         env:
+        {{- if .Values.ingressController.enabled }}
```

Comment:
Should there be an option to override this setting? What if there is a use case where Kong should be deployed with Admin API enabled along with Ingress controller?

Reply:
I agree, any suggestions on how to handle that ? Maybe:
`values.yml`

```
ingressController:
  admin:
     enabled: true   # <--- Enable admin in the controller deployment
     unique: true     # <--- Enable the admin in the controller deployment only
```

Reply:
Actually this is a bug.
If `ingressController.enabled` is true and `admin.useTLS` is set to to true, the behavior will be a weird.

Reply:
Any suggestions on how we should manage this form a Helm users perspective ? For me, documenting the incompatibility seems the way to go for now, but I'm open to any solution you might have on mind

Reply:
My recommendation would be to remove this if clause and let the Admin API be configured using the existing settings. We can recommend the user to disabled Admin API when running in the Ingress controller mode but if the need be, it  should be possible to enable it.

Let's not clobber Ingress Controller logic inside here. Ingress Controller really should ideally be a drop-in in addition to an existing Kong setup.

Thoughts?

Reply:
I removed this, so we should warn people in the docs .